keps/sig-storage/1682-csi-driver-skip-permission/README.md

```diff
@@ -0,0 +1,236 @@
+before bind mounting the volume inside container; however, if the volume
+does not support these operations (such as NFS), then the change is still attempted. 
+This results in errors being reported to the user if the volume doesn't 
+support these operations.
+
+### Goals
+
+ - Allow CSIDrivers to opt-in to volume ownership changes.
+
+### Non-Goals
+
+ - This does not attempt to update existing CSIDrivers themselves.
+
+## Proposal
+
+We propose that the `CSIDriver` type include a field that defines if the volume 
+provided by the driver supports changing volume ownership. This will be enabled
+with a new feature gate, `CSIVolumeSupportFSGroup`.   
+
+### Risks and Mitigations
+
+- The CSIDriver objects will need to be redeployed after this field is introduced if the desired behavior is modified.
+- If a cluster enables the `CSIVolumeSupportFSGroup` feature gate and then this feature gate is disabled,
+such as due to an upgrade or downgrade, then the cluster will revert to the current behavior of examining
+volumes and attempting to apply volume ownerships and permissions based on the defined `fsGroup`.
+
+## Design Details
+
+Currently Volume permission and ownership change is done automatically for every volume. 
+As part of this proposal we will change the algorithm that modifies volume ownership and permissions
+for CSIDrivers to check the new field, and skip volume ownership modifications if it is found to be
+`false`.
+
+When defining a `CSIDriver`, we propose that `CSIDriver.Spec` be expanded to include a new field entitled 
+`SupportsFSGroup` which can have following possible values:
+
+ - `OnlyRWO` --> Current behavior. Attempt to modify the volume ownership and permissions to the defined `fsGroup` when the volume is 
+ mounted if accessModes is RWO.
+ - `Never` --> New behavior. Attach the volume without attempting to modify volume ownership or permissions.
+ - `Always` --> New behavior. Always attempt to apply the defined fsGroup to modify volume ownership and permissions.
+
+```go
+type SupportsFsGroup string
+
+const(
+    OnlyRWO SupportsFsGroup = "OnlyRWO"
+    Always SupportsFsGroup = "Always"
+    Never SupportsFsGroup = "Never"
+)
+
+type CSIDriverSpec struct {
+    // SupportsFSGroup ← new field
+    // Defines if the underlying volume supports changing ownership and 
+    // permission of the volume before being mounted. 
+    // If set to Always, SupportsFSGroup indicates that 
+    // the underlying volume supports volume ownership and permission
+    // changes, and the filesystem will always be modified to match the defined
+    // fsGroup.
+    // If set to Never, then the volume will be mounted without modifying
+    // the volume's ownership or permissions.
+    // Defaults to OnlyRWO, which results in the volume being examined
+    // and the volume ownership and permissions attempting to be updated
+    // if fsGroup is defined and accessModes is RWO.
```

Comment:
nit: When does this happen?

Reply:
I've updated this to be more explicit:

> ...permissions attempting to be updated only when the PodSecurityPolicy's fsGroup is explicitly defined and the PersistentVolumes's accessModes is RWO.

I'm leaving this conversation open for now. Let me know if this is sufficient, or if it should be clearer.